agent/provider/src/execution/exeunit_instance.rs

```diff
@@ -34,6 +34,13 @@ impl ExeUnitInstance {
         let mut command = Command::new(&binary_path);
         command.args(args).current_dir(working_dir);
 
+        #[cfg(unix)]
+        {
+            // executes the command in its own process group
+            use ya_utils_process::CommandSetSid;
+            command.setsid();
+        }
+
```

Comment:
Could we keep provider free from system dependent code inside compilation switches??
In this case we could for example implement this trait for all systems (in process/src/lib.rs), but on non unix system it would resolve to empty implementation.

This call in provider could look like this:
```
command.args(args).current_dir(working_dir).new_process_group();
```

Reply:
@nieznanysprawiciel That was also my thought. But I'd personally expect this to also work on Windows (and that's not necessary - we're using `JobObject`s https://github.com/golemfactory/yagna/blob/e59a9ac9ca97922db0caabb25a708541d9257a78/exe-unit/src/process/win.rs)

Reply:
I wasn't talking only about functionalities. For me it is ugly, that we expose system level details here.
We need abstraction level that deals with differences between systems. _Provider Agent_ is on higher level and should only use abstraction.
Compilation switches like this should be hidden behind this abstraction.

Reply:
I've implemented it as requested (one of my previous iterations). 

I'd agree with you if there existed an implementation for all target platforms. Currently it just hides the fact that it doesn't do what is says (on Windows) and each use of this trait needs to be accompanied by a _doesn't work on Windows_ comment